File: CS/XtraGridTutorial.Step3/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DataLayer;
using System.Diagnostics;
using DevExpress.XtraEditors;
using DataLayer.Data.Classes;
using DevExpress.Utils;
using DevExpress.XtraEditors.Repository;

namespace XtraGridTutorial {
    public partial class MainForm : XtraForm {
        private UnitOfWork uow;
        public MainForm()
        {
            InitializeComponent();
            uow = new UnitOfWork();
            SetDataSource();
            InitControls();
        }

        private void barButtonSave_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            uow.Commit();
        }

        private void barButtonReload_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            uow.ReloadContext();
            SetDataSource();
        }

        private void SetDataSource()
        {
            accountBindingSource.DataSource = uow.GetData<Account>();
            categoryBindingSource.DataSource = uow.GetData<Category>();
            transactionBindingSource.DataSource = uow.GetData<Transaction>();
        }
        private void InitControls()
        {
            InitImageComboBox(repositoryItemImageComboBox1);
            InitImageComboBox(repositoryItemImageComboBox2);
        }

        private void InitImageComboBox(RepositoryItemImageComboBox item)
        {
            item.Items.AddEnum(typeof(TransactionType));
            for (int i = 0; i < item.Items.Count; i++)
            {
                item.Items[i].ImageIndex = i;
            }
        }
    }
}

File: VB/XtraGridTutorial.Step3/MainForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DataLayer
Imports System.Diagnostics
Imports DevExpress.XtraEditors
Imports DataLayer.Data.Classes
Imports DevExpress.Utils
Imports DevExpress.XtraEditors.Repository

Namespace XtraGridTutorial
    Partial Public Class MainForm
        Inherits XtraForm

        Private uow As UnitOfWork
        Public Sub New()
            InitializeComponent()
            uow = New UnitOfWork()
            SetDataSource()
            InitControls()
        End Sub

        Private Sub barButtonSave_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonSave.ItemClick
            uow.Commit()
        End Sub

        Private Sub barButtonReload_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonReload.ItemClick
            uow.ReloadContext()
            SetDataSource()
        End Sub

        Private Sub SetDataSource()
            accountBindingSource.DataSource = uow.GetData(Of Account)()
            categoryBindingSource.DataSource = uow.GetData(Of Category)()
            transactionBindingSource.DataSource = uow.GetData(Of Transaction)()
        End Sub
        Private Sub InitControls()
            InitImageComboBox(repositoryItemImageComboBox1)
            InitImageComboBox(repositoryItemImageComboBox2)
        End Sub

        Private Sub InitImageComboBox(ByVal item As RepositoryItemImageComboBox)
            item.Items.AddEnum(GetType(TransactionType))
            For i As Integer = 0 To item.Items.Count - 1
                item.Items(i).ImageIndex = i
            Next i
        End Sub
    End Class
End Namespace
